kraken/src/models/manual/operations.rs

```diff
@@ -386,3 +390,146 @@ impl ManualService {
         Ok(service_uuid)
     }
 }
+
+#[derive(Patch)]
+#[rorm(model = "ManualHttpService")]
+struct InsertManualHttpService {
+    uuid: Uuid,
+    name: String,
+    domain: Option<String>,
+    ip_addr: IpNetwork,
+    port: i32,
+    base_path: String,
+    tls: bool,
+    sni_require: bool,
+    user: ForeignModel<User>,
+    workspace: ForeignModel<Workspace>,
+}
+
+impl ManualHttpService {
+    /// Manually insert a http service
+    ///
+    /// This function will store the raw data given by the user
+    /// and add it to the aggregations.
+    ///
+    /// The [`HttpService`]'s uuid will be returned.
+    #[allow(clippy::too_many_arguments)]
+    pub async fn insert(
+        executor: impl Executor<'_>,
+        workspace: Uuid,
+        user: Uuid,
+        name: String,
+        domain: Option<String>,
+        ip_addr: IpAddr,
+        port: NonZeroU16,
+        base_path: String,
+        tls: bool,
+        sni_require: bool,
+    ) -> Result<Uuid, rorm::Error> {
```

Comment:
Protocol is missing

Reply:
in case we are dealing with quick, we mustn't assume tcp

Reply:
Which option would you prefer?
1. a boolean flag named something like `is_tcp` or `is_udp`
2. using `PortProtocol` which contains other variants besides TCP and UDP
3. create a new enum
This choices concerns the schema exposed to the API, the aggregator's method and the manual insert model

Reply:
No strong opinion between 2 and 3, no fan of 1

Reply:
went for 2 now